sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -124,13 +125,16 @@ abstract class DataType extends AbstractDataType {
 object DataType {
 
   private val FIXED_DECIMAL = """decimal\(\s*(\d+)\s*,\s*(\-?\d+)\s*\)""".r
+  private val CHAR_TYPE = """char\(\s*(\d+)\s*\)""".r
+  private val VARCHAR_TYPE = """varchar\(\s*(\d+)\s*\)""".r
```

Comment:
A question, do we need to support default length if user not give a special value ?

Reply:
We don't. We can set a default length/max length in followups.